### ISSUE TITLE
Add HTML tidying as an option

### ISSUE THREAD

**Issue by asomervell** [gohugoio/hugo#7190]
https://gohugo.io/news/0.69.0-relnotes/

Given the hugo templating engine creates whitespace hell in output now that  `resources.PostProcess` exists it would be awesome to have a similar hook to tidy the output HTML.

Here's a great Go lib that does it perfectly https://github.com/yosssi/gohtml

**Comment by bep** [comment]
You know you can minify the output?

**Comment by asomervell** [comment]
Yes. Not really the point though when you're design focused and looking at your DOM output. I know the javascript world has destroyed that but one of the nicest things about hugo is outputting simple, classic static html.

Thanks for all the work you do on hugo 🙏🏽

**Comment by bep** [comment]
You're right, the HTML output from Go templates leave much to be desired ...

Note that I think this should be implemented in the same place as we do output minification. `PostProcess` is something ... different (not sure what ... yet).

**Comment by fvbommel** [comment]
I've written an initial implementation of this: #7825

I'd love some feedback.

**Comment by jwflory** [comment]
Hi, I was migrating Jekyll sites where I use the handy [jekyll-tidy](https://github.com/apsislabs/jekyll-tidy) plug-in over to Hugo. Tidied HTML output was something I missed in Hugo and would make it much easier to design and work on my sites! A big 👍🏻 for seeing this one in a future release. 🤞🏻

**Comment by bep** [comment]
This is actually harder than it seems (I have looked at it). For one, github.com/yosssi/gohtml has some bugs in it and it's not very fast.

**Comment by jwflory** [comment]
I noticed a [forum thread](https://discourse.gohugo.io/t/revisiting-tidy-option/22685?u=jwf) where someone shared a number of other libraries, and also gave the idea of adding it a C library in Go using `hugo-extended`. Perhaps there are other ways to approach this than with `yosssi/gohtml`?

**Comment by bep** [comment]
> Perhaps there are other ways to approach this than with yosssi/gohtml?

Certainly not with a C library -- a feature must be really must have before we take that path.

**Comment by williamhaley** [comment]
As an inelegant workaround I do something like the following.

```
# build.sh

hugo

npx prettier@3.2.5 "public/**/*.html" --parser html --write
npx prettier@3.2.5 "public/**/*.json" --parser json --write
```

**Comment by BBaoVanC** [comment]
Even with the existence of newline chomping with `{{-`/`-}}`, it could still be improved as including a partial will throw its contents at 0 indentation and not really look "tidy".

Also, shouldn't issues that were "accepted" by being added to a milestone not be part of the stale job? I would've thought it would only be for issues discussed but not really confirmed/accepted.
